src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -1002,6 +1002,12 @@ void Terminal::_WriteBuffer(const std::wstring_view& stringView)
         _AdjustCursorPosition(proposedCursorPosition);
     }
 
+    // Notify UIA of new text.
+    // It's important to do this here instead of in TextBuffer, because here you have access to the entire line of text,
+    // whereas TextBuffer writes it one character at a time via the OutputCellIterator.
+    static std::wstring newline{ L"\n" };
+    _buffer->GetRenderTarget().TriggerNewTextNotification(std::wstring{ stringView } + newline);

```

Comment:
### Fun fact

Found some interesting behavior here when playing with different shells...
| Shell | Output when you type <kbd>a</kbd> + <kbd>b</kbd> + <kbd>c</kbd> |
|--|--|
| Ubuntu | `a` ... `b` ... `c` |
| PowerShell | `a` ... `ab` ... `abc` |
| PowerShell (disabled PSReadLine) | `a` ... `b` ... `c` |

As shown above, PowerShell actually rewrites the entire input buffer with each keyboard event! Why? Idk. But that made it particularly frustrating when I was trying to suppress the "local echo".

Fortunately, it seems like PowerShell has been aware of this issue for some time, because _if PowerShell detects that you are using a screen reader, it disables PSReadLine_!

So in your local testing, make sure you open your screen reader (or accessibility insights) first, _then_ open a PowerShell instance. That way, PowerShell will detect that you are using a screen reader and disable PSReadLine automatically (as well as output some text saying that it's done just that).